llvm/unittests/ADT/APFloatTest.cpp

```diff
@@ -560,6 +560,17 @@ TEST(APFloatTest, FMA) {
     EXPECT_EQ(-8.85242279E-41f, f1.convertToFloat());
   }
 
+  // Regression test for failing the `assert(!carry)` in
+  // `addOrSubtractSignificand` and normalizing the exponent even when the
+  // significand is zero if there is a lost fraction.
+  {
+    APFloat f1(-1.4728589E-38f);
```

Comment:
I assume more test values are required to hit the cmpGreaterThan, capLessThan, and cmpEqual cases